### ISSUE TITLE
Not clear to newcomers that source link is not supported with Mono debugger

### ISSUE THREAD

**Issue by tomeverin** [dotnet/sourcelink#561]
I'm new to source link. I've been working on a Xamarin.Forms project, and was attempting to debug nuget packages using source link. It wasn't working; spent quite some time trying to troubleshoot before finding the yellow box at the top of https://docs.microsoft.com/en-us/xamarin/xamarin-forms/internals/sourcelink

> Visual Studio 2019 supports Source Link for the .NET debugger but does not currently support Source Link for the Mono debugger. Therefore, you can use Source Link to debug UWP apps, but not Android or iOS app.

It would be useful to include this information in the sourcelink readme to hopefully save folks some time. 

Additionally; enhancement? Add support for mono debugger.

**Comment by jfichtner** [comment]
Is this still true? Support for Source Link was implemented in VS for Mac 2 years ago and still doesn't exist in VS for Windows? This is incredibly disheartening for someone who leads a team of Xamarin devs using VS for Windows! Will this ever be supported? Also seems strange that this particular issue is still marked as "Open" and "New issue" with zero responses in over 2 years.

**Comment by Tyrap** [comment]
We run a Xamarin application on the Android emulator where most of the code is included thru NuGet Packages (which are made by other teams). We are unable to debug this which gives us an undoable development experience.  Will this feature ever be implemented or are there any other ways to debug NuGet packages in an Xamarin application running on the device emulator?

**Comment by kkolodziejczak** [comment]
@Tyrap I have been working with NuGet packages and Xamarin for some time now. I can tell You that It is possible to debug NuGet Packages. The thing is that packages need to be build locally, and You do not even need source link for this. If you have source files locally, build packages locally, You can add the output directory as a source for packages, override NuGet versions for the local ones. If You run the app that way, you will get full debug experience.

**Comment by Tyrap** [comment]
Not working on my machine. I can build a package locally (with portable PDB files). When I consume it within my Xamarin android project and start the debugger I can see that the library is loaded with symbols. However, when I put a break point just before the call into the library, and want to step into the method, the first time I hit F11 nothing happen (would expect it gets into the method) and the second time I hit F11 it just steps over the method to the next statement.

If I do the same thing but reference the library directly in stead of using the NuGet Package it works fine. In that case I also see the library is loaded with symbols but this time when I hit the F11 it will step into the library method as what I would expect.

I am using Visual Studio 2022

**Comment by tmat** [comment]
@chuckries

**Comment by kkolodziejczak** [comment]
@Tyrap True, I forgot about one thing. It does work because of ReSharper. If You have it You need to set: `Options → Tools → Debugger → External Sources → Navigation to Sources` _check: Use sources from symbol files when available_.

**Comment by Tyrap** [comment]
I have Resharper with the option set but still it won't step into the method. I also tried it with the Rider IDE. same thing. it won't step into the method.

**Comment by kkolodziejczak** [comment]
Have you unchecked in VS options, `Debugging → Enable Just My Code`? 
Also have selected `Debugging → Symbols → Load all modules, unless excluded`, and specified `Cache symbols in this directory`?

**Comment by WoZ** [comment]
@kkolodziejczak I assume that your response is related to VS for Windows. VS for Mac doesn't have such fields (Just My Code) and so on.
